apps/condo/domains/common/components/settings/CommonSettingsContent.tsx

```diff
@@ -0,0 +1,129 @@
+import { Table } from 'antd'
+import { ColumnsType } from 'antd/es/table'
+import React, { useCallback, useMemo, useState } from 'react'
+
+import { LocaleContext, useIntl } from '@open-condo/next/intl'
+import { Button, Modal, Radio } from '@open-condo/ui'
+
+interface ISettingsPopupContentProps<T> {
+    value: T
+    onChange: (value: T) => void
+}
+
+interface ILanguageSettingPopupContentProps extends ISettingsPopupContentProps<string> {
+    possibleLocales: { [locale: string]: string }
+}
+
+interface ISettingsTableItem {
+    title: string
+    valueCell: JSX.Element
+    modalContentComponent: React.FC<ISettingsPopupContentProps<unknown>>
+    value: unknown
+    onChange: { (data: unknown): void }
+    props: { [key: string]: unknown }
+    onSave: { (data: unknown): void }
+}
+
+type TCurrentSettingModalData = Omit<ISettingsTableItem, 'valueCell'>
+
+const LanguageSettingPopupContent: React.FC<ILanguageSettingPopupContentProps> = ({
+    value,
+    onChange,
+    possibleLocales,
+}) => {
+    return (
+        <div>
+            {Object.entries(possibleLocales).map(([locale, localeName]) => (
+                <div key={`locale=${locale}`}>
+                    <Radio
```

Comment:
Could that be done with RadioGroup component?

Reply:
Yes.
Refactored.